### ISSUE TITLE
Proovread substring information may be incorrect

### ISSUE THREAD

**Issue by thefantasticdron** [BioInf-Wuerzburg/proovread#71]
Hello Thomas,

I think I have found an instance where the substring information provided in the header line for proovread corrected reads may be incorrect. I have enclosed a file containing two reads - the first one in the original uncorrected read and the second the corrected read outputted by proovread.

[reads.txt](https://github.com/BioInf-Wuerzburg/proovread/files/394630/reads.txt)

As you can see, the substring information written by proovread indicates that the corrected read corresponds with the first 3274 bases of the uncorrected read. The uncorrected read has a length of 3280 bases. Thus, the substring information states that the last 6 bases "ACCATT" of the uncorrected long read should not align with the corrected read. However, if we look at the last 6 bases of the corrected long read, we see that the exact same bases "ACCATT" also appears. Thus, the substring indicated by proovread is 6 bases shorter that it should be and the slice should then be "SUBSTR:0,3280". I have also found similar cases like this in other reads as well.

Am I interpreting this information correctly?

Thank you,
Sean

**Comment by thackl** [comment]
Hi Sean,

the problem here is that the substring information corresponds to the 
coordinates in the corrected consensus, not the original raw read.

Short example: Consider a raw read with two insertions due to pacbio 
error. First step is consensus from Illumina, which should remove the 
insertions, next step is quality based trimming. Even if the entire read 
has high quality, the substring will only be as long as the untrimmed 
consensus.

```
raw:        AATcATAgGAA
untrimmed:  AATATAGAA
trimmed:    AATATAGAA
```

The same seems to be true for your reads. Just align the two reads and 
count the number of gaps on query and reference. There should be 6 gaps 
more in the reference, accounting for the "missing" bases in substring info.

Cheers
Thomas

**Comment by thefantasticdron** [comment]
Ah I understand. Would it be possible, in future releases, to modify or include substring information for the original raw read? Currently I'm working on a project that uses proovread and I was hoping to use the substring information provided for the corrected reads to determine which segments of the uncorrected read were unable to be corrected.

**Comment by thackl** [comment]
Unfortunately, adding this information would require some rather 
sophisticated changes to proovread's work flow. I actually gave that 
some thought in the past, but did not come up with a good solution. 
Consensus and trimming are modularized steps, and there is no easy way 
to propagate the information, in particular, since it would have to be 
applied in a more or less dynamic matter. One would have to transpose 
coordinates for example based on a cigar string, which accounts for 
insertions as well as deletion, and keep track of those changes over 
several, independent iterations. Therefore, in case I needed to get the 
exact coordinates of uncorrected regions, I ended up using some form of 
back alignment of the corrected read to the raw read, rather than 
implementing a complex back tracking system.

That probably won't help you a lot, sorry...

Cheers
Thomas
